### DISCUSSION TITLE
Javascript deferreed import

### DISCUSSION THREAD

**Discussion by Vorkosigan76** [rails/jsbundling-rails#86] [Q&A]
In order to load a js library, you need to add an import request to _application.js_ like this
`import "@fortawesome/fontawesome-free/js/all.js"`

However some Javascript libraries require a deferred import such as this
`<script src="all.js" defer></script>`
The reason for this is that this code executes when loaded and requires the page to be fully loaded.

=> Is there a way to do that with jsbundling?

**Comment by justin808** [comment]
@Vorkosigan76 could you just have another file, like `fontawesome.js` with your one-liner import and then use that name with the view helper? 

```erb
<%= javascript_include_tag "fontawesome", defer: true %>
```

**Reply by Vorkosigan76** [reply]
Sure but it is ugly. I would like to include it in the bundler (that is the purpose of the bundler.

And technically I can have all JS from the bundler deferred... but same comment.

**Comment by dhh** [comment]
You're options are either to defer the whole bundle or to separate out fontawesome with its own file and defer that. But I'd just defer the whole bundle. That's also the new form for ESM in the browser. Everything included as a module is deferred by default.

**Reply by Vorkosigan76** [reply]
Yes that's what I did.
